packages/@aws-cdk/aws-stepfunctions-tasks/lib/athena/start-query-execution.ts

```diff
@@ -93,22 +94,28 @@ export class AthenaStartQueryExecution extends sfn.TaskStateBase {
 
     policyStatements.push(
       new iam.PolicyStatement({
-        actions: ['s3:AbortMultipartUpload',
-          's3:CreateBucket',
-          's3:GetBucketLocation',
-          's3:GetObject',
+        actions: ['s3:CreateBucket',
           's3:ListBucket',
+          's3:GetBucketLocation',
+          's3:GetObject'],
+        resources: ['*'], // Need * permissions to create new output location https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+      }),
+    );
+
+    policyStatements.push(
+      new iam.PolicyStatement({
+        actions: ['s3:AbortMultipartUpload',
           's3:ListBucketMultipartUploads',
           's3:ListMultipartUploadParts',
           's3:PutObject'],
-        resources: [this.props.resultConfiguration?.outputLocation ?? '*'], // Need S3 location where data is stored https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
+        resources: [this.props.resultConfiguration?.outputLocation?.bucketName ? 'arn:aws:s3:::' + this.props.resultConfiguration?.outputLocation?.bucketName + '/' + this.props.resultConfiguration?.outputLocation?.objectKey + '/*' : '*'], // Need S3 location where data is stored or Athena throws an Unable to verify/create output bucket https://docs.aws.amazon.com/athena/latest/ug/security-iam-athena.html
       }),
     );
 
     policyStatements.push(
       new iam.PolicyStatement({
         actions: ['lakeformation:GetDataAccess'],
-        resources: [this.props.resultConfiguration?.outputLocation ?? '*'], // Workflow role permissions https://docs.aws.amazon.com/lake-formation/latest/dg/permissions-reference.html
+        resources: ['*'], // State machines scoped to output location fail and * permissions are required as per documentation https://docs.aws.amazon.com/lake-formation/latest/dg/permissions-reference.html
```

Comment:
This resource does not get scoped to outputLocation. The only document I found that does not grant wildcard permissions is on page 78 of https://docs.aws.amazon.com/lake-formation/latest/dg/lake-formation-dg.pdf The default behaviour is to grant * access as found here: 
https://docs.aws.amazon.com/lake-formation/latest/dg/permissions-reference.html https://docs.aws.amazon.com/lake-formation/latest/dg/getting-started-setup.html
https://docs.aws.amazon.com/lake-formation/latest/dg/lake-formation-dg.pdf (page 21, 23, 27, 130, 150, 152, 153)